lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -1415,14 +1415,20 @@ void ConstraintSystem::print(raw_ostream &out) const {
   for (auto tv : typeVariables) {
     out.indent(2);
     Type(tv).print(out, PO);
+    std::vector<StringRef> properties;
     if (tv->getImpl().canBindToLValue())
-      out << " [lvalue allowed]";
+      properties.push_back("lvalue");
```

Comment:
We have talked offline about making this an enum + exhaustive switch to make it easier to add new "allowed" options.